### DISCUSSION TITLE
Build all the pull request created against any branch.

### DISCUSSION THREAD

**Discussion by jeran7x** [community/community#26990] [Actions]
<p>So the thing is we want to verify if the PR that is merged to a target branch does not break the build.</p>
<p>So we want to run a actions job to build all PR that are created against any branch on our repo</p>
<p><strong>Example 1 : </strong></p>
<p>master &lt;---- BRANCH_1.       </p>
<p>we need to merge this BRANCH_1 with master and run our build steps before we can merge the PR created against it.</p>
<p><strong>Example2 : </strong></p>
<p>BRANCH_DEV  &lt;-------- BRANCH_XX</p>
<p>We also need to merge this BRANCH_XX to BRANCH_DEV and run our build steps before we can merge the PR.</p>
<p>My current workflow file is </p>
<pre><code>on:
  push:
  pull_request:
- "**"
</code></pre>
<p>But when i do this there are multiple build jobs triggered. Can someone help me understand this case and point me the right way to do this.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/jeran7x">@jeran7x</a> ,</p>
<p>If your build jobs are triggered at the same time, please check if there are any status check required in branch protection rules on the target branch, please refer to official <a href="https://help.github.com/en/github/administering-a-repository/enabling-required-status-checks" rel="nofollow noopener">doc</a> for more details.<br>
Edit the source branch, it will trigger 2 workflows based on your workflow yaml, one is push event, the other is ‘pull_request: synchronize’ event. It’s recommended to check the github context of the workflow for confirmation.</p>
<p><img src="https://user-images.githubusercontent.com/52772761/181101081-1facc1ba-b58f-4ada-b633-def049cda24a.png" alt="PRinfo.png" data-base62-sha1="8k7G4i2AV3Lwy8huawaXBbP30GA" width="472" height="414"><br>
If the jobs are triggered one by one, please check if there’s a commit submit with personal access token to source branch, it will trigger a new ‘pull_request:synchronize’ event workflow.</p>
<p>pull_request workflow will not help to merge the code automatically, if you check the github context, you will find the affected ref is ‘/refs/pull/prnumber/merge’ which is a temporary&amp;fake merged branch.<br>
Unless you click the ‘Merge pull request’ button in pull request, or have code commit push during workflow, the code will not be merged into target branch.</p>
<p>pull_request event doesn’t accept ‘- “**”’,only accept <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-event-pull_request" rel="nofollow noopener">activity types</a>, <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags" rel="nofollow noopener">branches/tags</a>, <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths" rel="nofollow noopener">paths</a>.</p>
<p>If you’d like to trigger PR workflow for any branch, you need to put the workflow yaml file in <strong>each source branch</strong> , code sample as below:</p>
<pre><code>on: [pull_request]
</code></pre>
<p>Thanks.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/jeran7x">@jeran7x</a> , thanks for your kudos mark, is there any other query for this ticket? </p>
